packages/nimble-components/src/table/specs/table-column-specs/table-column-file-size.md

```diff
@@ -0,0 +1,294 @@
+# File Size Table Column
+
+## Overview
+
+The file size table column will provide a way to visualize numeric data that represents a file size (or other binary data size). It will automatically convert the raw value to use an appropriate display unit.
+
+### Background
+
+[Work Item](https://ni.visualstudio.com/DevCentral/_workitems/edit/2526114)
+
+### Non-goals
+
+-   Configurable number of decimal places
+-   Configurable units (e.g. display all values in KB)
+-   Grouping by value ranges/buckets
+
+### Features
+
+-   Converts the byte value to largest unit (e.g. KB, MB, GB) that results in a value of >=1
+-   Option to use 1024-based units (e.g. KiB) or 1000-based units (e.g. KB)
+-   Displays up to one decimal place
+-   Sorts/groups by numeric value (number of bytes) rather than display string
+-   Localized number and unit strings
+
+### Risks and Challenges
+
+### Prior Art/Examples
+
+Size column in [SLE Files grid](https://dev.lifecyclesolutions.ni.com/files/)
+
+Windows explorer (details view)
+
+---
+
+## Design
+
+Below is an example of how the `nimble-table-column-file-size` would be used within a `nimble-table`:
+
+```HTML
+<nimble-table>
+    <nimble-table-column-file-size field-name="size">Size</nimble-table-column-file-size>
+</nimble-table>
+```
+
+### API
+
+`nimble-table-column-file-size`
+
+-   `field-name` - name of the record field containing the data. Data must be a `number` byte count.
+-   `unit-type` - "binary" (KiB, MiB, GiB, TiB, PiB) or "decimal" (KB, MB, GB, TB, PB); defaults to "decimal"
```

Comment:
What's influencing our decision to default to `decimal`?

Reply:
KB/MB/GB seem much more prevalent KiB/MiB/GiB. For example, all of the prior art examples I gave, plus some below use the decimal-type units:
GitHub
![image](https://github.com/ni/nimble/assets/7282195/e5faf281-9fb8-4c6c-94fc-a18c06ab8343)
Gmail
![image](https://github.com/ni/nimble/assets/7282195/1376e8f2-4ff6-446a-affe-ccaee3aafc83)
npmjs.com
![image](https://github.com/ni/nimble/assets/7282195/4298458f-50d8-4ecd-995c-381c693a731f)

I couldn't find any examples of KiB/MiB/GiB being used in the wild. Granted, it's not clear if every example I found was actually using a 1000 conversion factor, but I don't have a reason believe otherwise.

Reply:
As a point of reference, it seems like Windows explorer (which shows units as KB, MB, GB), uses a conversion factor of 1024 (although sometimes the math really looks like they are using 1025, which is a bit perplexing).

In this example from the "Properties" window for a file, you can see that the KB is not the bytes divided by 1000:
![image](https://github.com/ni/nimble/assets/20542556/13811999-ff3d-47f6-90e3-52009131ef1a)

Reply:
This is not as straightforward as I thought, as it seems that 1024 is a better default conversion for file sizes, even though most places incorrectly use the decimal units (KB, etc).